.github/workflows/go.yml

```diff
@@ -8,25 +8,31 @@ on:
       - go.sum
       - justfile
 
+permissions:
+  contents: read
+
 jobs:
   lint:
     runs-on: ubuntu-latest
-    container: ghcr.io/linkerd/dev:v39-go
+    container: ghcr.io/linkerd/dev:v42-go
     steps:
       - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
+      - run: git config --global --add safe.directory "$PWD" # actions/runner#2033
       - run: just go-lint --verbose --timeout=10m
 
   fmt:
     runs-on: ubuntu-latest
-    container: ghcr.io/linkerd/dev:v39-go
+    container: ghcr.io/linkerd/dev:v42-go
     steps:
       - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
+      - run: git config --global --add safe.directory "$PWD" # actions/runner#2033
       - run: just go-fmt-check
 
   unit-test:
     runs-on: ubuntu-latest
-    container: ghcr.io/linkerd/dev:v39-go
+    container: ghcr.io/linkerd/dev:v42-go
     steps:
       - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac
+      - run: git config --global --add safe.directory "$PWD" # actions/runner#2033
```

Comment:
probably a bit out of scope but why do we need this now? pure curiosity from my side.

Reply:
Without this, the lint fails when the code was checked out with a recent git version. Something related to permissions:
https://github.com/actions/runner/issues/2033
Failure example:
https://github.com/linkerd/linkerd2-proxy-init/actions/runs/7061243421/job/19223102061